### PR TITLE
[SPARK-44892] Add official image Dockerfile for Spark 3.3.3

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?
Add Apache Spark 3.3.3 Dockerfiles.
- Add 3.3.3 GPG key
- Add .github/workflows/build_3.3.3.yaml
- ./add-dockerfiles.sh 3.3.3
- Add version and tag info

### Why are the changes needed?
Apache Spark 3.3.3 released:
https://spark.apache.org/releases/spark-release-3-3-3.html

### Does this PR introduce _any_ user-facing change?
Docker image will be published.

### How was this patch tested?
Add workflow and CI passed

### CODE REVIEW

**Comment by wangyum** [comment]
cc @Yikun

File: tools/template.py [hunk 1]
```diff
@@ -28,6 +28,8 @@
     "3.3.1": "86727D43E73A415F67A0B1A14E68B3E6CD473653",
     # issuer "viirya@apache.org"
     "3.3.2": "C56349D886F2B01F8CAE794C653C2301FEA493EE",
+    # issuer "yumwang@apache.org"
+    "3.3.3": "F6468A4FF8377B4F1C07BC2AA077F928A0BF68D8",
```
Comment (Yikun) [inline tools/template.py:32]: seems wrong and failed to check public key:

```
13.28 + gpg --keyserver hkps://keys.openpgp.org --recv-key F6468A4FF8377B4F1C07BC2AA077F928A0BF68D8
13.29 gpg: keybox '/tmp/tmp.yZmBm6Lueo/pubring.kbx' created
14.21 gpg: key A077F928A0BF68D8: new key but contains no user ID - skipped
14.21 gpg: Total number processed: 1
14.21 gpg:           w/o user IDs: 1
14.21 + gpg --batch --verify spark.tgz.asc spark.tgz
15.90 gpg: Signature made Fri 04 Aug 2023 04:18:57 PM UTC
15.90 gpg:                using RSA key F6468A4FF8377B4F1C07BC2AA077F928A0BF68D8
15.90 gpg:                issuer "yumwang@apache.org"
15.90 gpg: Can't check signature: No public key
```

Comment (wangyum) [reply tools/template.py:32]: It seems can't add keys from `hkps://keys.openpgp.org`:
```
#19 21.44 + gpg --keyserver hkps://keys.openpgp.org --recv-key F6468A4FF8377B4F1C07BC2AA077F928A0BF68D8
#19 21.49 gpg: keybox '/tmp/tmp.0eI3yRt1g0/pubring.kbx' created
#19 22.59 gpg: key A077F928A0BF68D8: new key but contains no user ID - skipped
#19 22.59 gpg: Total number processed: 1
#19 22.59 gpg:           w/o user IDs: 1
#19 22.59 + gpg --keyserver hkps://keyserver.ubuntu.com --recv-keys F6468A4FF8377B4F1C07BC2AA077F928A0BF68D8
#19 23.27 gpg: /tmp/tmp.0eI3yRt1g0/trustdb.gpg: trustdb created
#19 23.28 gpg: key A077F928A0BF68D8: public key "Yuming Wang (CODE SIGNING KEY) <yumwang@apache.org>" imported
#19 23.37 gpg: Total number processed: 1
#19 23.37 gpg:               imported: 1
#19 23.38 + gpg --batch --verify spark.tgz.asc spark.tgz
#19 29.95 gpg: Signature made Fri 04 Aug 2023 04:18:57 PM UTC
#19 29.95 gpg:                using RSA key F6468A4FF8377B4F1C07BC2AA077F928A0BF68D8
#19 29.95 gpg:                issuer "yumwang@apache.org"
#19 29.98 gpg: Good signature from "Yuming Wang (CODE SIGNING KEY) <yumwang@apache.org>" [unknown]
```

Comment (wangyum) [reply tools/template.py:32]: It seems that the key should be uploaded via: https://keys.openpgp.org/upload.
```
pub   rsa4096 2023-08-04 [SC]
      F6468A4FF8377B4F1C07BC2AA077F928A0BF68D8
uid                      Yuming Wang (CODE SIGNING KEY) <yumwang@apache.org>
sub   rsa4096 2023-08-04 [E]
-----BEGIN PGP PUBLIC KEY BLOCK-----

mQINBGTMm/ABEACejvAu4Dc74oIy3VGdz4g/axvoIEyj1q7bQYW+vJhdiWY2/yK/
U+s10es2WFHL6Vq3wjdhiVwlPWLBICkxPTIGcSBan6Qlqwhepnv27YJpTy/B/C6/
wgQC//9GkifOyxOIxKU+5WNMyGZOSXBcQvHkwcxCrVmt/Hz0mNG/jKkXc8EpU/hI
dY83G6tJMgar54N+24H0NsY7pfU/9NOno6Klnn3IzOw/tLM+V5GS9un4u6D/XG15
bvT9+c234CzV2q5Zo+sDrqTE7qHfY5aqQU9WaEFCLw/n+lu39Hpgi843mw+IgNMs
hltxOvg6vbTY50LGpqcUJSXQOka5jLNVQVv3k9i9DndkVO9AU98FnMP3sK3ZFmlH
YA/4iMmZ0+41KAgadqLfqn7d19uqLZgAu88RHVUeDOZA2LoXLzjn7TrGq9kD3WRZ
st1cDV9ESxzr9X3vvhuz3JNGKUQzhaq0Mw4HWwbUWVVT3MCxI+wDNBKaN8R76PfV
ISX50y146OJ1BwHq5C6i9IEyxpTTih67z8UpNOnwL38ekwxmgJAklsprt32HejsP
8ac3zmufEDyn2zJ+7/wWHei9eW2ZxUXMbY30xFRnYAAYgaE7HY5kQqwk7P9obvMb
hSTtUVTJ9+4SgaEBABgjbziQpFB/WxAF280EnIIqX0TibuS8LC9//w7SYQARAQAB
tDNZdW1pbmcgV2FuZyAoQ09ERSBTSUdOSU5HIEtFWSkgPHl1bXdhbmdAYXBhY2hl
Lm9yZz6JAlEEEwEIADsWIQT2RopP+Dd7TxwHvCqgd/kooL9o2AUCZMyb8AIbAwUL
CQgHAgIiAgYVCgkICwIEFgIDAQIeBwIXgAAKCRCgd/kooL9o2JfOD/9ZQRH3KgSB
mEYadvVfn0NzSJ9AwJBXvOHfO5IPEZ4T/R1+UGSPxJqWsnCNAGR/NU1YHgbj/31k
CB/Taep8S7PoQaj3c8/84DsPRRUgYuLILDYTo6wrgvk4g1rHy6u49+wiUY3gCwV/
uUpggfTPeqya6HkNOgNWXXUoLWKmpWqJYoAUFUBvNr+fdyac21QdbK7zSCyVZU70
Ov2YXya82f0NrCiUbOAfmQ9oEKVb0PDoOvzd0Db6cJ/7bii849/1/gWMZQDWmRnm
rYcD+JxKgMgOMp5SZgmu5AD7htbsF+D34D6snwkWXDzfMYqVsvsL4NE1ENm7qSLG
U6V9PAsRFCVKUg8xAI3vR5HX3Pvp6e4kz3zm56GMY9knHQFzaw7HdNaPQnujLG4Q
NqSxHXBOsJ01AJXj6eZTHANA/j5cjqrBISsxuvEH7HNgWO7/b2V17UsjUUIWG5SZ
IgtgqfEPonDh1EXwerXHr9D0uKGjYUlwnUQLlSa4f1yY15GinEvPkmQOTu8ddZaU
Sl/5vxmiYbzC2iIPJMHcwxb0Df1Ynp7GKhP/LOHNH+ROB97q2be+ZgSZKgqHBops
zBN93/FhjAEDwzVuMmySyofp5KmElg/Tox0xAgxyV3zWsUji6hon3KoAqLocoHJv
IIbLn4koHhM4ewy5BN9yeGH4C5Xc9NC0ObkCDQRkzJvwARAA4BxnWU48R+wSTE4u
9oXpXOAYwriOxt3al17NrcbH7j73y4U+BXbN2P2xGYjyz6PoETAtAP6EKokSn4ei
3jvSwVdy+XQa7T5vLiCTzNjEE9paTHLasQ10dS9+j1ow4WIV9L8NRCVQWBYlFqCd
6l/O0d1OFJhXb/8BfrEzVBPBR36Ee32ZrDXhJ06h1u7nPJVc4L0A5US5RJX2RXEB
UhbXEJfhOaSRANujL6h70DLp6GFkevlFH/CFwmrNvML8AF7sUFkcNdSP8HVIRbU2
KHclRSetkFvvIALXTcVG9PPz0/TIucwpkMFtzhi4U8r0Z4krNHjgs5lhjxrrI2j0
aBGywpAMG1nTe4rIOUio4p7ze+Be0pmbx2Zl0HWxbRmtNvFKTpGHpP0E3IsnxxpZ
q6U8vOGPvVxWIZVdOfUHMR2FCHsRoi6e8Vqw0bvhc7+7+cw9t5A2dXejG/j9jQ++
F2lwRFVlzxKbLooKIQecUaBn2O+Mxo/RlKqHFjmRE+VEQiQ6jRtFESfj5fpYgeuy
XuMGTDdG4dlOXZx/GaEqbpNilY2CK9Zal6JpVFKT9GlzeITLlRURMa/3589mI4mZ
VcmVa6AiM3B2lfljaZJGrTFntl+2hUILJsQJ+t7iH04+im00dpVFwXHA5ji+yM20
4C57hNFSEEE+QFaGF5FTq5fHEekAEQEAAYkCNgQYAQgAIBYhBPZGik/4N3tPHAe8
KqB3+Sigv2jYBQJkzJvwAhsMAAoJEKB3+Sigv2jYjnQP/i1T/JFdoUcyNoFWqisy
I/EbOPwMCsBJpDttd7JiW7yWKWdawdjDjQym5FKY5hResj7nox9fObUTNVzOBHsV
PWbIypP1i4uBQrjrg/klOhV2ltoq9bBMgaijzDcmDVf5kehThBfiwNwrfMpHgPhl
raU+fmF97vhfc0NDJKIkiq57wcQyU5RMkzitiZKg2i6sZwS3AZpCuGoScyczDaGV
HvIcJHG1OrXf6q+sQ13k5BU6Q7pqvewyJbv8SI8gnAWb+tlUQGIcP2/9B0D6EAOq
hFiM0rqIrY67lKluly+1/QEPTE8wcnSjxm0k9bGHeD+p0mBxILvXNhi1q03ROhzf
7L74MSjtwbZUoeQJ/lzMLFVcNTq+k4IAXs2fLnMu2Qsuue9HHq0W4xLcHYm9CqoV
17y2VGtK1XxZd4JIYQP/ZuRt0r6zOhRBOa8aThcg6Y6p8M18sRI/rWeadsmbJ3Sf
qmgGvDUTnmYv0yi+jNiTBNn002sxRZUz543xINd/j4J+iBlX+mn5IPA7b94MRL06
rCe7HD9pJgGh1/hsJnn20rpL7rRKQlkG2cuOf6V475ws+F16+DbjnOUbMF0JIKqw
gDOpXbrcjWwiB29RAh83kbtM6QvXAnyMhEAjFPC/Sz4znXMNLoRcKUtMGx7mELWo
a+TeRisx2Fp5t+/vM5rHYFdf
=CUri
-----END PGP PUBLIC KEY BLOCK-----
```

After uploading the key. You will receive a verification email:
image


```
yumwang@G9L07H60PK ~ % gpg --keyserver hkps://keys.openpgp.org --recv-key F6468A4FF8377B4F1C07BC2AA077F928A0BF68D8 
gpg: key A077F928A0BF68D8: "Yuming Wang (CODE SIGNING KEY) <yumwang@apache.org>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1

yumwang@G9L07H60PK ~ % gpg --batch --verify  spark-3.3.3-bin-hadoop3.tgz.asc spark-3.3.3-bin-hadoop3.tgz                                                       
gpg: Signature made 六  8/ 5 00:18:57 2023 CST
gpg:                using RSA key F6468A4FF8377B4F1C07BC2AA077F928A0BF68D8
gpg:                issuer "yumwang@apache.org"
gpg: Good signature from "Yuming Wang (CODE SIGNING KEY) <yumwang@apache.org>" [unknown]
gpg: WARNING: The key's User ID is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: F646 8A4F F837 7B4F 1C07  BC2A A077 F928 A0BF 68D8
```

**Review by Yikun — APPROVED** [review]
LGTM

**Comment by wangyum** [comment]
Thank you @Yikun . Merged to master.

**Pull request closed without merging**